lhotse/supervision.py

```diff
@@ -1,14 +1,68 @@
 import logging
 from dataclasses import dataclass
 from itertools import islice
-from typing import Any, Callable, Dict, Iterable, List, Mapping, Optional, Sequence
+from typing import Any, Callable, Dict, Iterable, List, Mapping, NamedTuple, Optional, Sequence
 
 from lhotse.serialization import Serializable
 from lhotse.utils import Seconds, asdict_nonull, exactly_one_not_null, fastcopy, \
     ifnone, index_by_id_and_check, \
     perturb_num_samples, split_sequence
 
 
+@dataclass(frozen=True, unsafe_hash=True)
+class AlignmentItem(NamedTuple):
```

Comment:
If `frozen` is True, is there a need to inherit from a tuple?
It is not modifiable any way when `frozen` is True.

Reply:
I thought about using tuples to reduce the memory overhead and the serialization overhead (store it as a list `['word', 0, 0.7]` rather than dict `{'symbol': 'word', 'start': 0, 'duration': 0.7}`). I did a simple check in jupyter and the named tuple variant is 2x faster to create and consumes 2x less memory...

Reply:
(I am concerned about this because there could be a lot of these objects around)